libraries/AP_SerialManager/AP_SerialManager.cpp

```diff
@@ -482,68 +519,293 @@ void AP_SerialManager::init()
+#endif
+#if !AP_ROBOTISSERVO_ENABLED
+                case SerialProtocol_Robotis:
+                    protocol_not_compiled_in_error("Robotis", i);
+                    break;
+#endif
+#if !AP_CAN_SLCAN_ENABLED
+                case SerialProtocol_SLCAN:
+                    protocol_not_compiled_in_error("SLCAN", i);
+                    break;
+#endif
+                case SerialProtocol_RCIN:
+#if AP_RCPROTOCOL_ENABLED
+                    check_no_duplicates_error(SerialProtocol_RCIN, i);
+#else
+                    protocol_not_compiled_in_error("RCIN", i);
+#endif
+                    break;
+
+#if !HAL_EFI_ENABLED
+                case SerialProtocol_EFI:
+                    protocol_not_compiled_in_error("EFI", i);
+                    break;
+#endif
+#if !HAL_MSP_ENABLED
+                case SerialProtocol_MSP:
+                case SerialProtocol_DJI_FPV:
+                case SerialProtocol_MSP_DisplayPort:
+                    protocol_not_compiled_in_error("MSP", i);
+                   break;
+#endif
+#if !AP_RANGEFINDER_ENABLED
+                case SerialProtocol_Rangefinder:
+                    protocol_not_compiled_in_error("Rangefinder", i);
+                    break;
+#endif
+#if !HAL_PROXIMITY_ENABLED
+                case SerialProtocol_Lidar360:
+                    protocol_not_compiled_in_error("Lidar360", i);
+                    break;
+#endif
+#if !AP_BEACON_ENABLED
+                case SerialProtocol_Beacon:
+                    protocol_not_compiled_in_error("Beacon", i);
+                    break;
+#endif
+#if !AP_DEVO_TELEM_ENABLED
+                case SerialProtocol_Devo_Telem:
+                    protocol_not_compiled_in_error("Devo_Telem", i);
+                    break;
+#endif
+#if !AP_OPTICALFLOW_ENABLED
+                case SerialProtocol_OpticalFlow:
+                    protocol_not_compiled_in_error("OpticalFlow", i);
+                    break;
+#endif
+#if !HAL_NMEA_OUTPUT_ENABLED
+                case SerialProtocol_NMEAOutput:
+                    protocol_not_compiled_in_error("NMEAOutput", i);
+                    break;
+#endif
+#if !AP_LTM_TELEM_ENABLED
+                case SerialProtocol_LTM_Telem:
+                    protocol_not_compiled_in_error("LTM_Telem", i);
+                    break;
+#endif
+                case SerialProtocol_RunCam:
+#if HAL_RUNCAM_ENABLED
+                    check_no_duplicates_error(SerialProtocol_RunCam, i);
+#else
+                    protocol_not_compiled_in_error("RunCam", i);
+#endif
+                    break;
+#if !HAL_HOTT_TELEM_ENABLED
+                case SerialProtocol_Hott:
+                    protocol_not_compiled_in_error("Hott", i);
+                    break;
+#endif
+#if !AP_SCRIPTING_ENABLED
+                case SerialProtocol_Scripting:
+                    protocol_not_compiled_in_error("Scripting", i);
+                    break;
+#endif
+                case SerialProtocol_CRSF:
+#if AP_RCPROTOCOL_CRSF_ENABLED
```

Comment:
Outside the case statement.  Similarly elsewhere.

Reply:
Fixed